### DISCUSSION TITLE
User (profile) tagging

### DISCUSSION THREAD

**Discussion by edokoevoet** [deso-protocol/dips#95] [Medium CIPs]
I'm preparing to set-up BitCloutNederland.com. However, this request is generic for BitClout<<country>>.com, and probably can be used beyond that. The idea behind it, is that it will improve user inflow and experience if they have a way to connect to their fellow country members as well.

The idea is that we would like to be able to tag users in such a way that we can share that data back into the blockchain. 

From a user perspective they would get to see: 1) global feed, 2) following feed and NEW 3) Dutchmen feed, 4) Dutch feed

The content in the 4) Dutch feed = function (all post by users in 3) Dutchmen feed; filtered on Dutch language)

Design questions related to this set-up: 
1) Who has the authority to fill, read and overwrite this tag ?
2) Does a user choice for a nationality restrict him to that same nationality when using other apps on top of BitClout? Or is it a value related to the application he's using that value with?
3) If user, app and possibly other entities can fill that tagging value, in what way can they than do that?

Guess we need a framework for tagging here. Obviously, there are way more use cases for tagging other than the nationality

**Comment by tijno** [comment]
Couple of thoughts

1 - see discussion about fn/ln - its relevant to this as well i think - eg whether increasing metadata on profiles is desired: https://github.com/bitclout/cips/discussions/24

2 - what your proposing is a protocol change and therefore would make this a large cip

3 - workaround: use flags in peoples profiles and search for that?

4 - pending solution: nfts with my suggested change would allow for badges which can be used as a way to tag users belonging to certain communities. See https://github.com/bitclout/cips/discussions/59

**Reply by edokoevoet** [reply]
Thank for your feedback Tijn. It got me thinking. Maybe it's better to approach this from a user preference than a user characteristic perspective.

Designing with the user in mind I would say:
* The user may decide to follow the "country feed" of one or more bitclout<<country>>.com sites
* He may change these preferences over time
* These decisions can be fully decoupled from the actual nationality of that person
* Users are only added to country feed after opt-in
* The above bullets are decoupled from whether or not elements of FN/LN and flag are added to his/her profile

Considerations:
* If the flag is there in the profile, is it ok to invite the user to the country feed? Via DM? How many times?
* If the user opts-in, is that an element that can be fed back into the bitclout blockchain from a privacy perspective?

Opt-in mechanism could look like:
* If you visit via bitclout<<country>>.com and follow bitcloutcountry profile then you get shown the country feed of that country

All in all I'm rethinking the approach: Is it about the application tagging the user, or is it about the user tagging an application and is it ok for the user that the application shares with the rest of the world that the user has tagged them

**Reply by tijno** [reply]
@edokoevoet 

> Designing with the user in mind I would say:

yes agree with those points. Could for implementation speed just setup a Pub Key for each country you support and have someone manually reclout specific posts relevant to that country.

> country feed after opt-in

maybe consider `@bitbadges` project if they have a feed per badge? Or use NFTs ones they implement the non-transferable request.

> fed back into the bitclout blockchain

Keep it off chain.

**Reply by edokoevoet** [reply]
Thank you for your feedback. We will do some DEV in the coming weeks and get back to how we dealt with these element in our set-up

**Comment by darian-parrish** [comment]
More generalized post tagging solution written up here: https://github.com/deso-protocol/dips/discussions/206
